feat: duplication removal dont overwrite (#32)
Agile-Arch-Angels/decentralized-tech-radar_dev: docs/dev_docs/merger.md

  * **What it does:** This function takes the provided filepath to an CSV file to open it, reads the specification file's header, and writes it to a byte array.

* `MergeCSV(filepaths []string, header string) error`
  * **What it is:** A public function taking two arguments: An Array of data type string, and a string. It returns nothing.
  * **What it does:** If Merged_file.csv already exists, this is removed. Then it uses `DuplicateRemoval()` to read each provided csv-file into a buffer. The buffer is then written to a Merged_file.csv file. These are read in the order of the file-paths provided.
  
* `MergeFromFolder(folderPath string) error`
  * **What it is**: A public function that takes one argument: A path to a folder, which in the default case, when adding the cache flag, is the cache folder itself.
  * **What it does**: If the cache folder exists, it reads all files from said folder, and appends them to cachePaths. It then checks whether or not cachePaths contain anything, and if so, merges the file with `MergeCSV(cachePaths)`.

* `DuplicateRemoval(buffer *bytes.Buffer, filepaths ...string) error`

>   * **What it is:**: A public function that takes a pointer to a byte buffer that the contains the merged specfiles' data without the duplicates and multiple filepaths to specfiles of type string.

Reviewer: ```suggestion
  * **What it is:**: A public function that takes a pointer to a byte buffer that contains the merged specfiles' data, without the duplicates, and multiple filepaths to specfiles of type string.
```

  * **What it does**: Goes through the files given line by line and calls `duplicateRemoval()` with each line.

* `duplicateRemoval(name, line string, buffer *bytes.Buffer, set map[string][]string) error`

>   * **What it is**: A private function that takes a name of the blip in that line, the line, the buffer for the merged specfiles, and the map for all the seen blips' names.

Reviewer: Now I'm unsure, does it take the entire line as well, or the rest of the line after the blip name?
```suggestion
  * **What it is**: A private function that takes a blip name from the given line, the line, the buffer for the merged specfiles, and the map for all the seen blips' names.
```

Author: It takes the whole line since that is what is written to the buffer.
